plugins/__init__.py

```diff
@@ -1093,3 +1094,76 @@ def is_base64_encoded(s: "string") -> "bool":
     except Exception:
         return False
     return True
+
+
+@plugin
+def hostname(fqdn: "string") -> "string":
+    """
+    Return the hostname part of the fqdn
+    """
+    return fqdn.split(".")[0]
+
+
+@plugin
+def netmask(prefixlen: "int") -> "std::ipv4_address":
+    """
+    Given the cidr, return the netmask
+    """
+    interface = ipaddress.ip_interface(f"255.255.255.255/{prefixlen}")
+    return str(interface.netmask)
+
+
+@plugin
+def cidr_to_prefixlen(addr: "std::ipv_any_interface") -> "int":
+    """
+    Return the prefixlen of the CIDR
+
+    For instance:
+        | std::print(cidr_to_prefixlen("192.168.1.100/24"))  -->  "24"
+    """
+    interface = ipaddress.ip_interface(addr)
+
+    return str(interface.network.prefixlen)
+
+
+@plugin
+def cidr_to_network_address(addr: "std::ipv_any_interface") -> "std::ipv_any_address":
+    """
+    Return the network address of the CIDR
+
+    For instance:
+        | std::print(cidr_to_network_address("192.168.1.100/24"))  -->  192.168.1.0
+    """
+    interface = ipaddress.ip_interface(addr)
+
+    return str(interface.network.network_address)
+
+
+@plugin
+def cidr_to_netmask(addr: "std::ipv_any_interface") -> "std::ipv_any_address":
+    """
+    Return the netmask of the CIDR
+
+    For instance:
+        | std::print(cidr_to_netmask("192.168.1.100/24"))    -->  255.255.255.0
+    """
+    interface = ipaddress.ip_interface(addr)
+
+    return str(interface.network.netmask)
+
+
+@plugin
+def ipindex(addr: "std::ipv_any_network", position: "number") -> "string":
+    """
+    Return the address at position in the network.
+    """
+    net = ipaddress.ip_network(addr, False)
+    return str(net[position])
+
+
+@plugin
+def add(addr: "std::ipv_any_address", n: "number") -> "std::ipv_any_address":
```

Comment:
```suggestion
def add_to_ip(addr: "std::ipv_any_address", n: "number") -> "std::ipv_any_address":
```

I would add some prefix, as this would produce unexpected results when used as `add(3,5)=="0.0.0.8"`

Reply:
this is still exactly the same?